src/Gateway.cc

```diff
@@ -0,0 +1,78 @@
+#include <cstring>
+#include "Gateway.h"
+
+namespace kickcat
+{
+    void Gateway::fetchRequest()
+    {
+        uint8_t frame[ETH_MTU_SIZE];
+        EthercatHeader  const* const header     = reinterpret_cast<EthercatHeader  const*>(frame);
+        mailbox::Header const* const mbx_header = reinterpret_cast<mailbox::Header const*>(frame + sizeof(EthercatHeader));
+
+        auto [frame_size, index] = socket_->recv(frame, ETH_MTU_SIZE);
+        if (frame_size < 0)
+        {
+            return;
+        }
+
+        // Check frame validity
+        // -> minimal size
+        constexpr ssize_t MIN_EXPECTED = sizeof(EthercatHeader) + sizeof(mailbox::Header);
+        if (frame_size < MIN_EXPECTED)
+        {
+            // Payload shall contain a valid mailbox EtherCAT frame
+            DEBUG_PRINT("Frame size is too small");
+            return;
+        }
+
+        // -> Frame command type
+        if (header->type != EthercatType::MAILBOX)
+        {
+            DEBUG_PRINT("Frame command is not MAILBOX but %d", header->type);
+            return;
+        }
+
```

Comment:
There is no info in the ETG8200 about what the gateway is suppose to return to the client in case of ill-formatted requests ?

Reply:
From 4.1
```
There is no error handling specified for the Mailbox Gateway functionality. A request to a non-existing
slave device can lead to no Response from the Master. The Device Configuration Tool has to handle
this e.g. with a timeout.
```